### ISSUE TITLE
[🐛 BUG] Blank chart after too much changes

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#1992]
## Description
If a chart's data changes a lot, the corresponding chart will go blank and will not return to its normal behavior without refreshing the page.

This is a regression. https://github.com/Avaiga/taipy-gui/issues/864

## How to reproduce

```python
from taipy.gui import Gui
from math import cos, exp

value = 10

page = """
Markdown
# Taipy *Demo*

Value: <|{value}|>

<|{value}|slider|>

<|{compute_data(value)}|chart|>
"""


def compute_data(decay:int)->list:
    return [cos(i/6) * exp(-i*decay/600) for i in range(10_000)]


Gui(page).run()
```

Change the slider a lot of times, play with it. The chart should at one point be blank and it is not possible to see the data even by changing the slider again. The user has to refresh the page.

## Expected behavior
When the chart goes blank, Taipy should try to make it work

![Screenshots](https://private-user-images.githubusercontent.com/98709993/255110336-1d543d3a-cf55-4c18-961c-5a24de9de206.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0ODUxODUsIm5iZiI6MTcyODQ4NDg4NSwicGF0aCI6Ii85ODcwOTk5My8yNTUxMTAzMzYtMWQ1NDNkM2EtY2Y1NS00YzE4LTk2MWMtNWEyNGRlOWRlMjA2LmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA5VDE0NDEyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiNWE3NjdiYjQ0YjIyODc3ODRlMTc4Y2Q2YjFiNWJlM2I4MGEyZGU5ZTBhN2FlNzI1MGJiMDMzNmZjYjU3MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ALTLTJzwfW106iK_RSzJ8APJ76P-aY0r-tCFb5C5GQM)

## Runtime environment
Problem was in Taipy GUI: 2.3.1
Solved in 3.0
Problem is now found in 4.0

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by wizaye** [comment]
hey @FlorianJacta  willing to work on this! Can you assign this to me?

**Comment by FlorianJacta** [comment]
This issue is not yet opened for contribution. Thank you!

**Comment by AlexandreSajus** [comment]
Fixed in 4.0.1.dev1
🎉
